src/manifest.c

```diff
@@ -1379,3 +1397,36 @@ int enforce_compliant_manifest(struct file **a, struct file **b, int searchsize,
 	}
 	return ret; // If collisions were found, so manifest is purely additive
 }
+
+/* The function splits a list of manifests into two lists:
+ * - full_manifests: this list will contain all the full manifests that were in the manifests list
+ * - iterative_manifests: this list will contain all the valid iterative manifests from the manifests
+ *   list, plus the full manifest of those bundles that didn't have a valid iterative manifest
+ *   A valid iterative manifest is one that can be used during an update, meaning that the iterative
+ *   manifest must match the "from version" provided, and that it belongs to an already installed bundle*/
+void filter_iterative_manifests(struct list *manifests, int version, struct list **full_manifests, struct list **iterative_manifests)
+{
+	/* Split the manifests into "iterative manifests" and "full manifests" */
+	struct list *list;
+	struct file *file;
+
+	for (list = manifests; list; list = list->next) {
+		file = list->data;
+		/* Keep only iterative manifests that are from the specified version and that are
+		 * already installed */
+		if (file->is_iterative == 1 && file->from_version == version  && is_tracked_bundle(file->bundlename)) {
```

Comment:
I have the feeling we can use file->from_version <= version

If we are updating from 20 to 30 and we have Manifest.bundle.I.10 it means there was no change in bundle between 10 and 30. So we can use it. @matthewrsj is there any corner case where this isn't valid?

But if we are updating from 10 to 30 and there is a Manifest.bundle.I.20 we can't use it, because we will miss files updated from 10 to 20.